### ISSUE TITLE
Setting `default = ""` in `get_list()` for a non-existent field returns empty character vector

### ISSUE THREAD

**Issue by ateucher** [r-lib/desc#138]
I believe this should return the default value set `""`:

``` r
library(desc)
desc <- desc(text = "Package: mypkg")

desc$get_list("foo", default = "")
#> character(0)
```

I believe the culprit is [here](https://github.com/r-lib/desc/blob/c8463062edf4dc5501c66d4e8f6019743fae3fa3/R/description.R#L899), calling `strpslit()` on the empty string returns `character(0)`.

I'm happy to do a PR if you'd like

<sup>Created on 2023-05-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by gaborcsardi** [comment]
> I'm happy to do a PR if you'd like

Yes, please!

**Comment by ateucher** [comment]
Ok great. Is the intention that `default` should be a length 1 character vector (including `""`), but nothing else?

**Comment by gaborcsardi** [comment]
Actually, now that I took a better look at this, I don't think that it is a bug. The default value is not value that is should be returned, but the value that we pretend to be in the config file. E.g.:

```
❯ desc::desc_get_list("foo", default = "list, item, three")
[1] "list"  "item"  "three"

❯ desc::desc_get_list("foo", default = "")
character(0)
```

AFAICT usethis also uses it this way: https://github.com/cran/usethis/blob/cee86ced02745b9552de72355b5d4152472fbfe8/R/description.R#L206

The documentation could be improved, though.

**Comment by ateucher** [comment]
Yeah, that's actually where I discovered it. I was working on r-lib/usethis#1837 (which replaces the code you linked to above) and was just surprised by the `character(0)` returned when looking for a non-existent key [here](https://github.com/r-lib/usethis/blob/50319800875e00f976e60f98e603c3145e8e6a15/R/proj-desc.R#L39-L44).

But if that's the expected value, then I'm happy with that.
